tests/test-find-remotes.sh

```diff
@@ -0,0 +1,225 @@
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+set -euo pipefail
+
+. $(dirname $0)/libtest.sh
+
+echo '1..1'
+
+# Create two upstream collection repositories with some example commits
+cd ${test_tmpdir}
+
+mkdir apps-collection
+ostree_repo_init apps-collection --collection-id apps-collection
+mkdir -p files
+pushd files
+${CMD_PREFIX} ostree --repo=../apps-collection commit -s "Test apps-collection commit 1" -b app1 --gpg-homedir=${TEST_GPG_KEYHOME} --gpg-sign=${TEST_GPG_KEYID_1} > ../app1-checksum
+${CMD_PREFIX} ostree --repo=../apps-collection commit -s "Test apps-collection commit 2" -b app2 --gpg-homedir=${TEST_GPG_KEYHOME} --gpg-sign=${TEST_GPG_KEYID_1} > ../app2-checksum
+popd
+${CMD_PREFIX} ostree --repo=apps-collection summary --update --gpg-homedir=${TEST_GPG_KEYHOME} --gpg-sign=${TEST_GPG_KEYID_1}
+
+mkdir os-collection
+ostree_repo_init os-collection --collection-id os-collection
+mkdir -p files
+pushd files
+${CMD_PREFIX} ostree --repo=../os-collection commit -s "Test os-collection commit 1" -b os/amd64/master --gpg-homedir=${TEST_GPG_KEYHOME} --gpg-sign=${TEST_GPG_KEYID_2} > ../os-checksum
+popd
+${CMD_PREFIX} ostree --repo=os-collection summary --update --gpg-homedir=${TEST_GPG_KEYHOME} --gpg-sign=${TEST_GPG_KEYID_2}
+
+# Create a local repository where we pull the branches from the two remotes as normal, using GPG.
+mkdir local
+ostree_repo_init local
+${CMD_PREFIX} ostree --repo=local remote add apps-remote file://$(pwd)/apps-collection --collection-id apps-collection --gpg-import=${test_tmpdir}/gpghome/key1.asc
+${CMD_PREFIX} ostree --repo=local remote add os-remote file://$(pwd)/os-collection --collection-id os-collection --gpg-import=${test_tmpdir}/gpghome/key2.asc
+
+${CMD_PREFIX} ostree --repo=local pull apps-remote app1
+${CMD_PREFIX} ostree --repo=local pull os-remote os/amd64/master
+
+${CMD_PREFIX} ostree --repo=local refs > refs
+assert_file_has_content refs "^apps-remote:app1$"
+assert_file_has_content refs "^os-remote:os/amd64/master$"
+
+${CMD_PREFIX} ostree --repo=local refs --collections | wc -l > refscount
+assert_file_has_content refscount "^0$"
+
+# Create a local mirror repository where we pull the branches *in mirror mode* from the two remotes.
+# This should pull them into refs/mirrors, since the remotes advertise a collection ID.
```

Comment:
I think we may need a way to mirror into `refs/heads` as before - otherwise this will be a backcompat issue since older versions of libostree won't know how to pull from `refs/mirrors` right?

Although I guess this requires using `remote add --collection-id`...so if I want to run a "transparent mirror", I'd omit the collection ID?